docs/adr/adr_backend_etl_and_hosting_for_StatViz.md

```diff
@@ -0,0 +1,237 @@
+# ADR: How to do etl in the backend for the DSEE100x project "StatViz"?
+
+Decision Deadline:
+
+Author: @pylipp, @HaGuesto
+
+## Status
+
+proposed
+
+## Context
+
+In the DSEE 100x project titled "StatViz", our goal is to construct data visualizations derived from Boxtribute's data. This will allow our users to showcase their contributions and impact to the public through various mediums, such as their websites or newsletters.
+
+Currently, data meant for these visualizations is planned to be served through a public GraphQL endpoint that is implemented using a stack of MySQL, Peewee, Flask, and Ariadne.
+
+## Problem Statement
+
+While we are committed to implementing visualizations, several questions related to data extraction, transformation, loading (ETL), hosting, and caching need to be addressed:
+
+- **Data Structure Evaluation**: Our existing data structure's compatibility needs assessment. Can we efficiently extract the necessary data for all the proposed visualizations using the present structure? Or is there a need to modify or extend our data schema, as suggested [here](https://github.com/boxwise/boxtribute/blob/master/docs/adr/adr_database-history.md)?
+
+- **Data Transformation**: Given our technology stack, what are the best practices to transform and serve data efficiently and rapidly?
+
+- **Data Format**: To ensure consistency and seamless integration between backend and frontend, we must define a standard format for returning the data. What should this format look like?
+
+## Decision Drivers
+
+1. **Non-Interference with Boxtribute's Performance**: Any solution we adopt should not compromise the responsiveness and efficiency of the main Boxtribute application.
+
+2. **Security**: No private data should be accessible to the public. We do not want to introduce new security risks.
+
+3. **Cost Efficiency**: Implementing the ETL process and serving the data should be done with minimal additional financial impact.
+
+4. **Server Response Latency**: The time it takes for the server to respond with the requested data should be minimized to ensure a seamless user experience.
+
+5. **Manageable Data Size**: The size of the returned data should be optimized so that frontend applications can perform operations like dynamic filtering and sorting without performance degradation.
+
+6. **Utilization of Existing Data**: We should target that we can already leveraging the data already present in Boxtribute.
+
+7. **Scalability**
+
+8. **Data Freshness**: There isn't a requirement for real-time data. A one-day lag in data freshness is acceptable, which might open up opportunities for batch processing or scheduled updates.
```

Comment:
one day lag is not as precise as an x hour lag. I'd also be interested in how much of a lock-in this requirement is. If we want to move to, say, a 3 hour lag, what's the difficulty / implication of that?